tests/scripts/thread-cert/sniffer_transport.py

```diff
@@ -148,11 +148,71 @@ def recv(self, bufsize):
         return bytearray(data), nodeid
 
 
+class MacFrame(ctypes.Structure):
+    _fields_ = [("buffer", ctypes.c_ubyte * 128),
+                ("length", ctypes.c_ubyte),
+                ("nodeid", ctypes.c_uint)]
+
+class SnifferVirtualTransport(SnifferTransport):
+    """ Virtual interface based implementation of sniffer transport. """
+
+    def __init__(self, nodeid):
+        self.Handle = None
+
+        # Load the DLL
+        self.Api = ctypes.WinDLL("otnodeapi.dll")
+        if self.Api == None:
+            raise OSError("Failed to load otnodeapi.dll!")
+
+        # Define the functions
+        self.Api.otListenerInit.argtypes = [ctypes.c_uint]
+        self.Api.otListenerInit.restype = ctypes.c_void_p
+
+        self.Api.otListenerFinalize.argtypes = [ctypes.c_void_p]
+
+        self.Api.otListenerRead.argtypes = [ctypes.c_void_p, ctypes.POINTER(MacFrame)]
+
+    def __del__(self):
+        if not self.is_opened:
+            return
+
+        self.close()
+
+    def open(self):
+        if self.is_opened:
+            raise RuntimeError("Transport is already opened.")
+
+        # Initialize a listener
+        self.Handle = self.Api.otListenerInit(0)
+
+        if not self.is_opened:
+            raise RuntimeError("Transport opening failed.")
+
+    def close(self):
+        if not self.is_opened:
+            raise RuntimeError("Transport is closed.")
+
+        self.Api.otListenerFinalize(self.Handle);
+        self.Handle = None
+
+    @property
+    def is_opened(self):
+        return bool(self.Handle is not None)
+
+    def recv(self, bufsize):
```

Comment:
What is the expected behavior of this function? In `sniffer.py` the `Sniffer._sniffer_main_loop` function calls it in a loop while `self._thread_alive.is_set()`. But if this is supposed to be a blocking function, until the transport is closed, what prevents it from getting stuck in `Sniffer.stop`? I seem to get into that state 'stuck' at the end of a run.